mcs/class/corlib/System.Security.Cryptography.X509Certificates/X509Helper.cs

```diff
@@ -49,6 +49,41 @@ internal static void InstallNativeHelper (INativeCertificateHelper helper)
 				Interlocked.CompareExchange (ref nativeHelper, helper, null);
 		}
 
+		public static X509CertificateImpl InitFromHandle (IntPtr handle)
+		{
+#if MONO_FEATURE_APPLETLS && ONLY_APPLETLS // ONLY_APPLETLS should not support any other option
+			return InitFromHandleApple (handle);
+#elif MONOTOUCH_WATCH
+			throw new PlatformNotSupportedException ();
+#else
+
+#if MONO_FEATURE_APPLETLS // If we support AppleTls, which is the default, and not overriding to legacy
+			if (System.Environment.IsMacOS && Environment.GetEnvironmentVariable ("MONO_TLS_PROVIDER") != "legacy")
+				return InitFromHandleApple (handle);
+#endif
+#if !MOBILE
+			return InitFromHandleCore (handle);
+#elif !MONOTOUCH && !XAMMAC
+			throw new NotSupportedException ();
+#endif
+#endif
+		}
+
+		static X509CertificateImpl Import (byte[] rawData)
+		{
+#if MONO_FEATURE_APPLETLS && ONLY_APPLETLS // ONLY_APPLETLS should not support any other option
+			return ImportApple (rawData);
+#elif MONOTOUCH_WATCH
+			throw new PlatformNotSupportedException ();
+#else
+#if MONO_FEATURE_APPLETLS
+			if (System.Environment.IsMacOS && Environment.GetEnvironmentVariable ("MONO_TLS_PROVIDER") != "legacy")
+				return ImportApple (rawData);
+#endif
+			return ImportCore (rawData);
+#endif
```

Comment:
*Cosmetic-Post-Landing*
Instead of using such system checks all over the place, we should add a new `static internal` helper method to `MonoTlsProviderFactory` - I'll clean that up later.